BF/TST: check PATH is in repo for `onyo tree PATH`; add `test_tree.py`

TobiasKadelka opened the pull request:
`onyo tree PATH` did test that the opdir is in the repository, but did not test if the PATH given as input is in the repository, which I fixed.

To verify that the fix does not introduce new bugs, I added `test_tree.py`, because it seems we never had any tests at all for this command.

Close #121

aqw commented on onyo/commands/tree.py:
```diff
@@ -18,18 +18,26 @@ def sanitize_directories(directories: list, opdir: str) -> list[str]:
-              file=sys.stderr)
+    if error_path or no_path:
+        if error_path:
+            print('The following paths are not in the Repository:\n' +
```
Nit: `Repository` should not be capitalized.

In general, I prefer when errors are explicit about what is wrong.

The issue is not "that the path is not in the repo". It's that all paths must be in the repo.

TobiasKadelka replied:
Thanks, I thought about a few ways of stating this more explicitly, please let me know what you think :)

aqw reviewed (commented):
Looks good overall. I left a few nits.

I did not review the tests, as I'm unsure about how `tree` should be implemented going forward (keep as-is, vs pass through like `config`, vs replace with a python implementation). But that ambiguity is not a blocker to these initial tests.

TobiasKadelka commented:
I updated the variable names and the output of `tree.py` and updated this PR!

I am not sure about tree as well, and we discussed the idea in the past to reimplement it in python.
But I think until then there is still reason to verify that the internal call of tree does the right thing.

And I want to state that I thought about these possible changes while implementing the tests, and I think when implementing something else, the tests will provide still reasonable test-szenarios to provide helpful feedback while I am updating `onyo tree` then :)

aqw commented on onyo/commands/tree.py:
```diff
@@ -9,27 +9,37 @@
     dirs = []
-    error_path = []
+    path_not_in_repo_error = []
+    no_directory_error = []
```
Sorry to nit. Please feel free to say if I'm descending into bike shedding.

I just see that these two vars basically do the same thing, yet the naming convention is completely different, which then (incorrectly) implies that they are quite far apart in utility.

Perhaps something similar to:
```
error_path_not_in_repo = []
error_path_not_dir = []
```

aqw replied:
I'm comfortable with the variables being separate. IMO, that's a matter of choice, regarding how specific the error messages should be. I have no problems with the direction you chose here.

+1 on the renaming for consistency. Thanks. :-)

TobiasKadelka commented:
I nitted the new variable names into the fabric of `tree.py`, thanks for looking into all the PRs! :)

Feel free to merge, when the tests where successful.

aqw reviewed (approved):
LGTM!
